src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -112,109 +132,143 @@ static ColliderCylinderInit sCylinderInit = {
 
 static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, MASS_IMMOVABLE };
 
-void func_80AE9A20(EnLiftNuts* this, PlayState* play) {
-    static s16 D_80AEBF60 = 0;
-    static s32 D_80AEBF64 = false;
+void EnLiftNuts_AddSharedMemoryEntry(EnLiftNuts* this, PlayState* play) {
+    static s16 sMinigameScore = 0;
+    static s32 sAddedSharedMemory = false;
 
-    if (!D_80AEBF64) {
-        func_800B6584(play, ACTOR_EN_GAMELUPY, &D_80AEBF60, sizeof(s16));
-        D_80AEBF64 = true;
+    if (!sAddedSharedMemory) {
+        Actor_AddSharedMemoryEntry(play, ACTOR_EN_GAMELUPY, &sMinigameScore, sizeof(s16));
+        sAddedSharedMemory = true;
     }
-    this->ptr_1EC = &D_80AEBF60;
+    this->minigameScore = &sMinigameScore;
 }
 
-void func_80AE9A80(EnLiftNuts* this, PlayState* play) {
-    static s16 D_80AEBF68 = false;
+void EnLiftNuts_FreeSharedMemoryEntry(EnLiftNuts* this, PlayState* play) {
+    static s16 sFreedSharedMemory = false;
 
-    if (!D_80AEBF68) {
-        func_800B6608(play, ACTOR_EN_GAMELUPY);
-        D_80AEBF68 = true;
+    if (!sFreedSharedMemory) {
+        Actor_FreeSharedMemoryEntry(play, ACTOR_EN_GAMELUPY);
+        sFreedSharedMemory = true;
     }
 }
 
-s32 func_80AE9AC4(EnLiftNuts* this, s32 arg1) {
-    static s32 D_80AEBF6C = 0;
+typedef enum {
+    /* 0 */ ENLIFTNUTS_AUTOTALK_MODE_CHECK_OFF,
+    /* 1 */ ENLIFTNUTS_AUTOTALK_MODE_SET_ON,
+    /* 2 */ ENLIFTNUTS_AUTOTALK_MODE_SET_OFF
+} EnLiftNutsAutoTalkMode;
 
-    switch (arg1) {
-        case 0:
-            if (D_80AEBF6C == 0) {
+s32 EnLiftNuts_AutoTalk(EnLiftNuts* this, EnLiftNutsAutoTalkMode mode) {
+    static s32 sIsAutotalkOn = false;
+
+    switch (mode) {
+        case ENLIFTNUTS_AUTOTALK_MODE_CHECK_OFF:
+            if (!sIsAutotalkOn) {
                 return true;
             }
             break;
 
-        case 1:
-            if (D_80AEBF6C == 0) {
-                this->unk_34E = 1;
-                D_80AEBF6C = 1;
+        case ENLIFTNUTS_AUTOTALK_MODE_SET_ON:
+            if (!sIsAutotalkOn) {
+                this->autotalk = true;
```

Comment:
Maybe do the same as above? I don't know if you think it looks weird, I'm sort of 50/50 on this one. Feel free to say no:
```suggestion
                this->autoTalk = true;
```